### ISSUE TITLE
FFTW installed with Spack 0.11.0 gets built without optimisations

### ISSUE THREAD

**Issue by boegel** [spack/spack#6999]
When playing around with Spack 0.11.0 to compare it with other software installation tools (for my upcoming talk https://fosdem.org/2018/schedule/event/installing_software_for_scientists/), I was benchmarking FFTW installations and noticed that the FFTW installed by Spack was *very* slow.

It turns out that FFTW is being installed without any optimisation flags, which is quite bad since the default for GCC is to compile with `-O0` (i.e. no optimisation), as opposed to other compilers like Intel which default to using `-O2`.

I consulted the Spack Slack channel, and @alalazo confirmed that he was seeing the same issue when using Spack 0.11.0 (but not current `develop`).

### Expected Result

* FFTW is built with optimisations
* FFTW benchmark runs in about 5 seconds

### Actual Result

* FFTW is built without optimisations
* FFTW benchmark runs in > 30 seconds

### Steps to reproduce the issue

```console
spack install fftw
spack load fftw@3.3.6-pl2
```

```console
curl -OL http://micro.stanford.edu/mediawiki/images/a/a9/Simple_example.tar
tar xfv Simple_example.tar
cd simple_example
sed -i'' 's/\(N[01] =\) [0-9]*/\1 8192/g' simple_example.c
gcc -O2 -march=native simple_example.c -lfftw3 -o simple_example -lm
time ./simple_example
```

### Information on your system

* CentOS 7.4 (system compiler is GCC 4.8.5)
* Intel E5-2680v3 (Haswell-EP) 2.5GHz, 64GB RAM
* Spack 0.11.0 (default configuration)

**Comment by alalazo** [comment]
Side note. A quicker way to reproduce this is to build the serial version only of FFTW:
```console
$ spack install fttw~mpi
```

**Comment by alalazo** [comment]
Looking at the revisions I suspect the bug was introduced in #4421 and solved in #6415 (which is not in the release). I searched for an open issue on this, but couldn't find it - probably I just don't know which keyword to look for. 

@becker33 Is it possible that:
https://github.com/spack/spack/blob/f962aba6ce85ba001bbe20f6f849478fdb9370c5/lib/spack/spack/build_systems/autotools.py#L184-L191

was setting empty `CFLAGS=''` environment variables and #6415 fixed it? Am I just missing context on a known issue?

**Comment by alalazo** [comment]
@tgamblin @boegel Closing, as 0.11.1 has been released
